### DISCUSSION TITLE
Trying the Markdown plugin

### DISCUSSION THREAD

**Discussion by laotala828** [umaranis/svelte-lexical#78] [General]
Hey, great job on the awesome work! I'm currently working on the markdown plugin and have run into a couple of issues. Here's the code I've been using. 

```ts
<script lang="ts">
	import { onMount } from 'svelte';
	import { getEditor } from 'svelte-lexical';
	import {
		registerMarkdownShortcuts,
		BOLD_STAR,
		HEADING,
		ORDERED_LIST,
		UNORDERED_LIST
	} from '@lexical/markdown';

	const editor = getEditor();

	onMount(() =>
		registerMarkdownShortcuts(editor, [BOLD_STAR, HEADING, ORDERED_LIST, UNORDERED_LIST])
	);
</script>
```

After running it, I encountered two errors. I managed to fix the first one involving the Prism error by adding the prism-svelte package. However, I'm a bit stuck on the second problem 
```
Uncaught Error: Unable to find an active editor state. State helpers or node methods can only be used synchronously during the callback of editor.update() or editorState.read().
    at getActiveEditorState (@lexical_markdown.js?v=59d6d3e9:6190:13)
    at $getSelection (@lexical_markdown.js?v=59d6d3e9:5861:23)
    at readEditorState (chunk-MJWQAJ5V.js?v=59d6d3e9:6815:12)
    at _EditorState.read (chunk-MJWQAJ5V.js?v=59d6d3e9:7812:12)
    at @lexical_markdown.js?v=59d6d3e9:9641:35
    at triggerListeners (chunk-MJWQAJ5V.js?v=59d6d3e9:6989:21)
    at commitPendingUpdates (chunk-MJWQAJ5V.js?v=59d6d3e9:6950:3)
    at chunk-MJWQAJ5V.js?v=59d6d3e9:7177:9
```
I could use some help figuring it out. Thanks!

**Comment by umaranis** [comment]
Hi @laotala828 ,

Do you have a repo where I can reproduce the issue?

**Reply by laotala828** [reply]
Here is the link to the PR to repo https://github.com/laotala828/svelte-lexical-markup/pull/1

**Reply by laotala828** [reply]
@umaranis thanks a lot!

**Comment by umaranis** [comment]
Hi @laotala828,

The following changes fix the issue in your markdown branch:

![SCR-20240424-jqzt](https://github.com/umaranis/svelte-lexical/assets/2135071/a413114b-c5dd-432b-95af-f38839b563d1)

**Reply by umaranis** [reply]
Would you submit a pull request for the Markdown plugin?

**Reply by laotala828** [reply]
Thank you so much! That works! 
Do you mind creating that PR? Honestly, I'm happy to give it a try myself, but I'm primarily a designer and I write shitty code. I wouldn't want to mess up your great work.

**Reply by umaranis** [reply]
No worries :)
